addons/columns/userscript.js

```diff
@@ -148,11 +148,22 @@ export default async function ({ addon, msg, console }) {
     toolbox.populate_(workspace.options.languageTree);
     // Reposition the toolbox, since it's likely our addon moved it.
     toolbox.position();
+
+    addClass();
+  }
+
+  function addClass() {
+    // Add class to allow editor-compact to handle this addon
+    if (addon.tab.editorMode === "editor") {
+      if (addon.self.disabled) document.querySelector("[class*='gui_tab-panel']").classList.toggle("sa-columns", false);
+      else document.querySelector("[class*='gui_tab-panel']").classList.toggle("sa-columns", true);
+    }
   }
 
   updateToolbox();
   addon.self.addEventListener("disabled", updateToolbox);
   addon.self.addEventListener("reenabled", updateToolbox);
+  addon.tab.addEventListener("urlChange", addClass);
```

Comment:
Is this enough? This won't react to actions such as changing the language of the editor.